From: Miroslav Benes <mbenes@suse.cz>
Subject: [PATCH] kgr: reserve a place in thread_struct for storing RIP on
 x86_64
Patch-mainline: Never, SUSE-Xen specific
References: fate#313296

In order to be able to patch __schedule() function with kGraft we need
to store proper RIP in switch_to macro and use it as a return address.
This ensures that a task will follow consistent path (with respect to
old/new universes) all the way in __schedule(). Otherwise it could be
possible that a task is woken up with the different code than it went
to sleep with. The RIP should be stored in thread_struct embedded to
task_struct. Reserve a place for that even for x86_64 arch.

Signed-off-by: Miroslav Benes <mbenes@suse.cz>
Automatically created from "patches.suse/kgr-0107-reserve-a-place-in-thread_struct-for-storing-RIP.patch" by xen-port-patches.py

--- a/arch/x86/include/mach-xen/asm/processor.h
+++ b/arch/x86/include/mach-xen/asm/processor.h
@@ -471,9 +471,7 @@ struct thread_struct {
 	unsigned short		fsindex;
 	unsigned short		gsindex;
 #endif
-#ifdef CONFIG_X86_32
 	unsigned long		ip;
-#endif
 #ifdef CONFIG_X86_64
 	unsigned long		fs;
 #endif
